docs/dev/adding_instruments/channels.rst

```diff
@@ -57,26 +60,111 @@ In order to add or remove programatically channels, use the parent's :meth:`~pym
             """Control the output voltage of this channel.""",
         )
 
-    class InstrumentWithChannels(Instrument):
+    class ExtremeVoltage5000(Instrument):
         """An instrument with channels."""
-        channels = Instrument.ChannelCreator(VoltageChannel, ("A", "B"))
+        output_A = Instrument.ChannelCreator(VoltageChannel, "A")
+        output_B = Instrument.ChannelCreator(VoltageChannel, "B")
 
 .. testcode:: with-protocol-tests
     :hide:
 
-    with expected_protocol(InstrumentWithChannels,
+    with expected_protocol(ExtremeVoltage5000,
         [("SOURceA:VOLT 1.23", None), ("SOURceB:VOLT?", "4.56")],
         name="Instrument with Channels",
     ) as inst:
-        inst.ch_A.voltage = 1.23
-        assert inst.ch_B.voltage == 4.56
+        inst.output_A.voltage = 1.23
+        assert inst.channels['B'].voltage == 4.56
+
+At instrument class instantiation, the instrument class will create an instance of the channel class and assign it to the class attribute name.
+Additionally the channels will be collected in a dictionary, by default named :code:`channels`.
+We can access the channel interface through that class name:
+
+.. code-block:: python
```

Comment:
I think it would be good to get these examples under protocol tests, too, so we notice if our behaviour would not match the docs anymore. You can do this with `.. testcode:: with-protocol-tests` sections as demonstrated in the rest of the file.

Reply:
I had the same thought, looking at the code, but forgot to mention it.

Reply:
Ok, I've added a protocol tests doctest that accomplishes the same thing as the python example.